syntax = "proto2";

enum PoplarOp {
  AvgPool = 0;
  AvgPoolGrad = 1;
  GroupNormGrad = 2;
  GroupNormInference = 3;
  GroupNormStatistics = 4;
  GroupNormTraining = 5;
  LstmLayerBwd = 6;
  LstmLayerFwd = 7;
  MaxPool = 8;
  MaxPoolGrad = 9;
  TruncatedNormal = 10;
  Relu = 11;
  ReluGrad = 12;
  TanhGrad = 13;
  Sigmoid = 14;
  SigmoidGrad = 15;
  Remap = 16;
  StatelessRandomUniform = 17;
  StatelessRandomUniformInt = 18;
  StatelessRandomNormal = 19;
  StatelessTruncatedNormal = 20;
  IpuInterCopy = 21;
  Dropout = 22;
  OneHot = 23;
  ReplicationFactor = 24;
  ReplicationNormalise = 25;
  ArgMin = 26;
  ArgMax = 27;
  PrintTensor = 28;
  StatefulGradientAccumulate = 29;
  StatefulGradientAccumulateAndAllReduce = 30;
  SelectScalarFromRows = 31;
  UpdateScalarInRows = 32;
  Fifo = 33;
  TopK = 34;
  UserOp = 35;
  RemapDeduce = 36;
  GRULayerBwd = 37;
  GRULayerFwd = 38;
  ReplicationIndex = 39;
  CodeletExpressionOp = 40;
  StatefulNoop = 41;
  CalcGfloatParams = 42;
  CastNativeToGfloat = 43;
  CastGfloatToNative = 44;
  MultiSlice = 45;
  MultiUpdate = 46;
  MultiUpdateAdd = 47;
  PipelineStatefulGradientAccumulate = 48;
  AllGather = 49;
  SuggestRecompute = 50;
  BlockRecompute = 51;
  ReduceScatter = 52;
  HostEmbeddingLookup = 53;
  HostEmbeddingUpdate = 54;
  StatefulGradientAccumulateWithMomentum = 55;
  StatefulGradientAccumulateWithMomentumAndAllReduceWithNorm = 56;
}
